packages/server/lib/webhook/checkr-webhook-routing.ts

```diff
@@ -0,0 +1,48 @@
+import crypto from 'node:crypto';
+
+import type { Config as ProviderConfig } from '@nangohq/shared';
+import { getLogger } from '@nangohq/utils';
+import type { WebhookHandler } from './types.js';
+import type { LogContextGetter } from '@nangohq/logs';
+
+const logger = getLogger('Webhook.Checkr');
+
+interface CheckrBody {
+    id: string;
+    object: string;
+    type: string;
+    created_at: string;
+    webhook_url: string;
+    data: Record<string, unknown>;
+    createdAt: string;
+}
+
+function validate(integration: ProviderConfig, headerSignature: string, rawBody: string): boolean {
+    if (!integration.custom?.['webhookSecret']) {
+        return false;
+    }
+
+    const signature = crypto.createHmac('sha256', integration.custom['webhookSecret']).update(rawBody).digest('hex');
+    return crypto.timingSafeEqual(Buffer.from(signature), Buffer.from(headerSignature));
+}
+
+const route: WebhookHandler = async (nango, integration, headers, body, rawBody, logContextGetter: LogContextGetter) => {
+    const signature = headers['x-checkr-signature'];
+
+    logger.info('received', { configId: integration.id });
+
+    if (!validate(integration, signature, rawBody)) {
+        logger.error('invalid signature', { configId: integration.id });
+        // TODO the verification should use the API key
+        //return;
```

Comment:
I'm not I get what is preventing you to return here?

Reply:
In testing there is no API key so we actually have no way to validate the webhook signature right now. Checkr will be forward only for now and that fits to the customers spec